### ISSUE TITLE
[Bug]: Blursed earth only spawning mobs from current biome?

### ISSUE THREAD

**Issue by qidydl** [FTBTeam/FTB-Modpack-Issues#493]
### Modpack

FTB OceanBlock

### Modpack version

1.11.0 (same problem was in 1.9.0)

### Describe the bug

A mob farm using Blursed Earth from Mini Utilities only seems to be spawning mobs according to the biome in which it is located, not all mobs. Given that the entire world is only ocean biomes, this makes it rather less useful because you only ever get fish, squids, and slimes. My world is on Normal difficulty, but switching to Hard had no effect, and it also doesn't seem to matter if I am right next to the farm or standing far away.

### Steps to reproduce

1. Make a dark room in the vicinity of the spawn island (mine is 8x8x5, or internal volume of 6x6x3)
2. Place Blursed Earth as the floor of the room
3. Automate killing the mobs and collecting drops (I used Industrial Foregoing's Mob Crusher and Mob Slaughter Factory running simultaneously to get both fluids and mob drops)
4. Observe you get nothing but fish, squid ink, and slime balls

### Expected behaviour

All mobs are spawned, including cows/pigs/sheep/chickens as well as skeletons/zombies, and mod mobs like blizzes

### Additional information

Direwolf20 built a very similar Blursed Earth mob farm in his FTB OceanBlock Let's Play series that was definitely spawning all mobs at the time, so this could be a recent regression. I don't know what version he was on when it was working for him, or whether he has altered the pack in any way, or if this behavior has started happening for him or not. I did not change the pack configuration.

**Comment by LuciusV** [comment]
There is probably related problem with blessed earth: if I'm rather far of initial spawn point (outer "blue" circle ocean zone) it doesn't spawn anything at all

**Comment by qidydl** [comment]
Doing some research... this is pretty speculative because my Java is rusty, but maybe someone who knows what they're doing can check it and run with it faster. It looks like OceanBlock v1.11.0 uses Mini Utilities v1.2.3 ([source](https://github.com/OneLemonyBoi/MiniUtilities/tree/e783a52c82d09a110575e4e182536124a8ee59c0/src/main/java/onelemonyboi/miniutilities)). In that version, the blessed/cursed/blursed earth blocks all have their own code so there could be differences between them. Blursed Earth appears to perform this check to determine whether to spawn a mob:

```java
        //can the mob actually spawn here naturally, filters out mobs such as slimes which have more specific spawn requirements but
        // still show up in spawnlist; ignore them when force spawning
        if (!EntitySpawnPlacementRegistry.canSpawnEntity(entry.type, world, SpawnReason.NATURAL, pos, world.rand)
                && MiniUtilities.blacklisted_entities.contains(entry.type))
            return null;
```

If I'm reading this right, only mobs that are "unnatural" spawns *and* blacklisted will be skipped, where `blacklisted_entities` is defined as:

```java
public static final ITag<EntityType<?>> blacklisted_entities = EntityTypeTags.getTagById(new ResourceLocation(MOD_ID, "blacklisted").toString());
```

It looks like this may be blank by default, but maybe OceanBlock is somehow populating it?

**Comment by qidydl** [comment]
Ok, maybe this is entirely on me, I figured out the problem. If I set up in a _Deep_ Warm Ocean biome, everything works just fine, but non-deep Warm Ocean is apparently completely different and doesn't allow anything to spawn. I would argue that this is confusing, especially for a mod pack where everything is ocean, but I understand if that's just a quirk of the biomes that can't be fixed, or maybe is even intentional.
